### PR TITLE
Move things relating to punishments to punishments.js

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

**Comment by bumbadadabum** [comment]
Fuck I fucked up so much uhh

**Comment by bumbadadabum** [comment]
Hope this is better

File: app.js [hunk 1]
```diff
@@ -190,11 +192,11 @@ fs.readFile(path.resolve(__dirname, 'config/ipbans.txt'), (err, data) => {
...
 			rangebans.push(data[i]);
-		} else if (!Users.bannedIps[data[i]]) {
-			Users.bannedIps[data[i]] = '#ipban';
+		} else if (!Punishments.bannedIps[data[i]]) {
+			Punishments.bannedIps[data[i]] = '#ipban';
 		}
 	}
-	Users.checkRangeBanned = Cidr.checker(rangebans);
+	Punishments.checkRangeBanned = Cidr.checker(rangebans);
 });
```
Comment (Slayer95) [inline app.js]: We probably want to do this in ``punishments.js``

File: app.js [hunk 2]
```diff
@@ -113,6 +113,8 @@ global.Ladders = require(Config.remoteladder ? './ladders-remote.js' : './ladder
 
 global.Users = require('./users.js');
 
+global.Punishments = require('./punishments.js');
```
Comment (Slayer95) [inline app.js]: We probably don't need this as a global...

File: users.js [hunk 3]
```diff
@@ -883,9 +872,9 @@ class User {
 			} else if (userType === '4') {
 				this.autoconfirmed = userid;
 			} else if (userType === '5') {
-				this.lock(false, userid + '#permalock');
+				Punishments.lock(userid, false, userid + '#permalock');
 			} else if (userType === '6') {
-				this.ban(false, userid);
+				Punishments.ban(userid, false);
```
Comment (Slayer95) [inline users.js]: This is a (bad) behavior change.

Comment (Zarel) [inline users.js]: I would prefer `Punishments.ban(user, noRecurse, userid)`

Comment (bumbadadabum) [inline users.js]: the user is the name used by permalocks etc? That one's not mandatory so I'd rather have that in the back.

File: app.js (hunk 2 above)
Comment (Zarel) [inline app.js]: How would you prefer, then?

File: users.js (hunk 3 above)
Comment (Zarel) [inline users.js]: no, `user` is the user object...

Comment (Zarel) [inline users.js]: i.e. this line would be `Punishments.ban(this, false, userid);`

File: commands.js [hunk 4]
```diff
@@ -1245,7 +1245,7 @@ exports.commands = {
 		if (userid !== toId(this.inputUsername)) this.add('|unlink|hide|' + toId(this.inputUsername));
 
 		this.globalModlog("LOCK", targetUser, " by " + user.name + (target ? ": " + target : ""));
-		targetUser.lock(false, userid);
+		Punishments.lock(targetUser, false);
```
Comment (Zarel) [inline commands.js]: you still need to pass `userid`...

Comment (Zarel) [inline commands.js]: Punishments.lock(targetUser, false, userid);

File: commands.js [hunk 5]
```diff
@@ -1333,7 +1333,7 @@ exports.commands = {
 
 		this.add('|unlink|hide|' + userid);
 		if (userid !== toId(this.inputUsername)) this.add('|unlink|hide|' + toId(this.inputUsername));
-		targetUser.ban(false, userid);
+		Punishments.ban(targetUser, false);
```
Comment (Zarel) [inline commands.js]: ditto here

**Comment by bumbadadabum** [comment]
ok did that

File: app.js (hunk 2 above)
Comment (Joimer) [inline app.js]: If the below checks are moved into punishments.js, then it can just be required in users. A bit convoluted with the current code, though.

Comment (bumbadadabum) [inline app.js]: Problem with that would be that they're used in both users.js and
commands.js. I guess I could add a Users.Prototype.ban() that just calls
Punishments.ban(this, etc, etc) but it's fine enough with the global I
think.

Comment (bumbadadabum) [inline app.js]: uhh, User.prototype I mean

Comment (Slayer95) [inline app.js]: That's not a problem. Just require it in ``commands.js`` as well.

**Comment by Zarel** [comment]
Good first step.

File: app.js (hunk 1 above)
Comment (Slayer95) [inline app.js]: This was a pretty uncontroversial change?
Why wasn't it done? Do you expect @Zarel's input for everything, @bumbadadabum ?

Comment (bumbadadabum) [inline app.js]: No I wanted to do it but Zarel merged it already, oh well. I did plan to clean everything up tomorrow, including this. And no, I do not expect Zarel's input for everything, but this PR was something he specifically asked me to do so I do want his input on what he wants from the things he asked me to do.

File: app.js (hunk 2 above)
Comment (Zarel) [inline app.js]: Just requiring it in Users makes it impossible to hotpatch, though... The entire reason we have all these globals is for ease of hotpatching...

File: app.js (hunk 1 above)
Comment (Zarel) [inline app.js]: Well, it's a slightly more in-depth refactor; I don't mind it being in a different commit.

Comment (Zarel) [inline app.js]: I asked him to rush a minimum refactor, so don't blame him for not getting it all done.

**Pull request merged**